docs/user_guide/source/engines/inline.rst

```diff
@@ -6,21 +6,50 @@ The Inline engine provides in-process communication between the writer and reade
 
 This engine is experimental, and is focused on the N -> N case: N writers share a process with N readers, and the analysis happens 'inline' without writing the data to a file or copying to another buffer. It has similar considerations to the streaming SST engine, since analysis must happen per step.
 
-To use this engine, you can either specify it in your xml config file, with tag ``<engine type=Inline>`` or set it in your application code:
+To use this engine, you can either specify it in your XML config file, with tag ``<engine type=Inline>`` or set it in your application code:
 
 .. code-block:: c++
 
     adios2::IO inlineIO = adios.DeclareIO("ioName");
     inlineIO.SetEngine("Inline");
+    inlineIO.SetParameters({{"writerID", "inline_write"}, {"readerID", "inline_read"}});
     adios2::Engine inlineWriter = inlineIO.Open("inline_write", adios2::Mode::Write);
-    io.SetParameters({{"writerID", "inline_write"}});
     adios2::Engine inlineReader = inlineIO.Open("inline_read", adios2::Mode::Read);
 
-Notice that unlike other engines, the reader and writer share an IO instance. Also, the ``writerID`` parameter allows the reader to connect to the writer.
+Notice that unlike other engines, the reader and writer share an IO instance. Also, the ``writerID`` parameter allows the reader to connect to the writer, and ``readerID`` allows writer to connect to the reader.
 
-For successful operation, the writer will perform a step, then the reader will perform a step in the same process. Data is decomposed between processes, and the writer can write its portion of the data like other Adios engines. When the reader starts its step, the only data it has available is that written by the writer in its process. To select this data in Adios, use a block selection. The reader then can retrieve whatever data was written by the writer.
+For successful operation, the writer will perform a step, then the reader will perform a step in the same process. Data is decomposed between processes, and the writer can write its portion of the data like other ADIOS engines. When the reader starts its step, the only data it has available is that written by the writer in its process. To select this data in ADIOS, use a block selection. The reader then can retrieve whatever data was written by the writer.
 
-Please refer to the examples or tests for the typical access pattern. Note, however, that the ``Get()`` method:
+.. note::
+ The inline engine does not support Sync mode for writing. In addition, since the inline engine does not do any data copy, the writer should avoid changing the data contents before the reader has read the data.
```

Comment:
Can `adios2::mode != adios2::mode::Sync` be enforced in code? (Sorry if that's already enforced, but I didn't see it if it is.)

Reply:
Resolved by unit test I didn't notice..

Reply:
So I think it's enforced as much as it can be. In the case of variables that aren't single values, an error is thrown if a PutSync is performed. In the case of single values, they're by default treated as sync, even if the user set the mode to deferred, right? So it seems like if I throw an error on those, then it will fail even when the user has actually selected deferred for single value variables.

Reply:
Oh somehow your last comment about the unit test didn't show up until after I added my comment